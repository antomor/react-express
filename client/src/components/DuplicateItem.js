import React from 'react';
import Paper from 'material-ui/Paper';
import { GridListTile } from 'material-ui/GridList';
import { withStyles } from 'material-ui/styles';
import grey from 'material-ui/colors/grey';
import Typography from 'material-ui/Typography';

const styles = theme => ({
  root: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginTop: theme.spacing.unit * 3,
    marginLeft: theme.spacing.unit * 3,
    backgroundColor: grey[500],
  }),
  itemKey: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginTop: theme.spacing.unit * 3,
    borderRadius: '10px'
  }),
  itemValue: theme.mixins.gutters({
    paddingTop: 8,
    paddingBottom: 8,
    marginTop: theme.spacing.unit * 3,
    textAlign: 'center',
    borderRadius: '10px'
  })
});

const DuplicateItem = props => {
  const { classes } = props;
  return (
    <GridListTile key={props.item.key} cols={3}>
      <Paper className={classes.root}> 
        <Paper className={classes.itemKey}>
          <Typography variant="subheading" gutterBottom>
            {props.item.key}
          </Typography>
        </Paper>
        <Paper className={classes.itemValue}>
          <Typography variant="subheading" gutterBottom>
            {props.item.value}
          </Typography>
        </Paper>
      </Paper>
    </GridListTile>
  )
}

export default withStyles(styles)(DuplicateItem);