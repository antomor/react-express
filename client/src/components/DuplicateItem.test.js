import React from 'react';
import DuplicateItem from './DuplicateItem';
import renderer from 'react-test-renderer';

describe('Duplicate item', () => {
  it('render', () => {
    const item = {
      key: 'aa',
      value: 1
    }
    const component = renderer.create(
      <DuplicateItem item={item}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

}); 
