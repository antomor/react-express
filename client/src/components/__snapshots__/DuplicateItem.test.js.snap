// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Duplicate item renders 1`] = `
<li
  className="MuiGridListTile-root-4"
>
  <div
    className="MuiGridListTile-tile-5"
  >
    <div
      className="MuiPaper-root-8 MuiPaper-shadow2-12 MuiPaper-rounded-9 DuplicateItem-root-1"
    >
      <div
        className="MuiPaper-root-8 MuiPaper-shadow2-12 MuiPaper-rounded-9 DuplicateItem-itemKey-2"
      >
        <h3
          className="MuiTypography-root-35 MuiTypography-subheading-42 MuiTypography-gutterBottom-52"
        >
          aa
        </h3>
      </div>
      <div
        className="MuiPaper-root-8 MuiPaper-shadow2-12 MuiPaper-rounded-9 DuplicateItem-itemValue-3"
      >
        <h3
          className="MuiTypography-root-35 MuiTypography-subheading-42 MuiTypography-gutterBottom-52"
        >
          1
        </h3>
      </div>
    </div>
  </div>
</li>
`;
